async.lua

```diff
@@ -0,0 +1,26 @@
+local areas = {}
+local core = core
+
+local safe_file_write = core.safe_file_write
+if safe_file_write == nil then
+    function safe_file_write(path, content)
```

Comment:
This looks sketchy. Does it write to `_G`? maybe not. Either way, this would make it clear that the local variable is assigned:

```Lua
safe_file_write = function(path, content)
```